torchrl/objectives/value/advantages.py

```diff
@@ -73,22 +88,64 @@ def is_functional(self):
         )
 
     @_self_set_grad_enabled
+    @dispatch_kwargs
     def forward(
         self,
         tensordict: TensorDictBase,
-        *unused_args,
-        params: Optional[List[Tensor]] = None,
-        target_params: Optional[List[Tensor]] = None,
+        params: Optional[TensorDictBase] = None,
+        target_params: Optional[TensorDictBase] = None,
     ) -> TensorDictBase:
-        """Computes the GAE given the data in tensordict.
+        """Computes the TDEstimate given the data in tensordict.
+
+        If a functional module is provided, a nested TensorDict containing the parameters
+        (and if relevant the target parameters) can be passed to the module.
 
         Args:
             tensordict (TensorDictBase): A TensorDict containing the data
                 (an observation key, "action", "reward", "done" and "next" tensordict state
                 as returned by the environment) necessary to compute the value estimates and the TDEstimate.
+                The data passed to this module should be structured as :obj:`[*B, T, F]` where :obj:`B` are
+                the batch size, :obj:`T` the time dimension and :obj:`F` the feature dimension(s).
+            params (TensorDictBase, optional): A nested TensorDict containing the params
+                to be passed to the functional value network module.
+            target_params (TensorDictBase, optional): A nested TensorDict containing the
+                target params to be passed to the functional value network module.
 
         Returns:
-            An updated TensorDict with an "advantage" and a "value_error" keys
+            An updated TensorDict with an advantage and a value_error keys as defined in the constructor.
+
+        Examples:
+            >>> from tensordict import TensorDict
+            >>> value_net = SafeModule(
+            ...     nn.Linear(3, 1), in_keys=["obs"], out_keys=["state_value"]
+            ... )
+            >>> module = TDEstimate(
+            ...     gamma=0.98,
+            ...     value_network=value_net,
+            ...     differentiable=False,
+            ... )
+            >>> obs, next_obs = torch.randn(2, 1, 10, 3)
+            >>> reward = torch.randn(1, 10, 1)
+            >>> done = torch.zeros(1, 10, 1, dtype=torch.bool)
+            >>> tensordict = TensorDict({"obs": obs, "next": {"obs": next_obs}, "done": done, "reward": reward}, [1, 10])
+            >>> _ = module(tensordict)
+            >>> assert "advantage" in tensordict.keys()
+
+        The module supports non-tensordict (i.e. unpacked tensordict) inputs too:
+
+        Examples:
+            >>> value_net = SafeModule(
```

Comment:
cc @tcbegley this is something a bit more interesting than TensorDictModule
This module is a regular nn.Module that reads from a tensordict.
You may have a some crazy combination of keys that your value network needs to read, but you'd still be able to use regular tensors with this.